### DISCUSSION TITLE
Внедрение плеера VLC

### DISCUSSION THREAD

**Discussion by trueromanus** [anilibria/anilibria-winmaclinux#121] [Обсуждения]
@WaffleLapkin @Koi-foo @X1mper @KnightMurloc
Друзья! Всем привет! Я залил новую версию в мастер в ней помимо всего прочего также есть полная интеграция для сборки с плеером VLC. Но пока чтобы не ломать ничего я ее выключил. Чтобы включить надо сделать следующее:
в файле src/AniLibria.pro надо поменять вот эти строчки:
```
#unix {
#    LIBS += -lvlc

#    INCLUDEPATH += /usr/include/
#    DEPENDPATH += /usr/include/

#    INCLUDEPATH += /usr/include/vlc/plugins
#    DEPENDPATH += /usr/include/vlc/plugins

#    CONFIG += buildwithvlc
#}
```
на вот эти:
```
unix {
    LIBS += -lvlc

    INCLUDEPATH += /usr/include/
    DEPENDPATH += /usr/include/

    INCLUDEPATH += /usr/include/vlc/plugins
    DEPENDPATH += /usr/include/vlc/plugins

    CONFIG += buildwithvlc
}
```
Но чтобы оно собралось стоит учитывать что требуется вначале установить пакеты с vlc, libvlc и libvlccore (последние два содержат заголовочные файлы).  
Как будет у Вас время пожалуйста проверьте чтобы оно собиралось и если нет тегайте меня в этом топике по ошибкам будем пытаться решить.  
На этом у меня все, как всегда заранее спасибо! 😊

**Comment by Koi-foo** [comment]
Ошибки выскакивают:
```
libvlc: nfs_parse_url_incomplete failed: 'Invalid URL specified'
libvlc: local stream 1 error: Cancellation (0x8)
libvlc: Could not convert timestamp 0 for FFmpeg
libvlc: peer error: No error (0x0)
libvlc: peer stream 1 error: Cancellation (0x8)
```

При закрытии видео:
```
[h264 @ 0x7fd2f53dcf00] get_buffer() failed
[h264 @ 0x7fd2f53dcf00] thread_get_buffer() failed
[h264 @ 0x7fd2f53dcf00] decode_slice_header error
[h264 @ 0x7fd2f53dcf00] no frame!
```

Работать это не мешает, видео загружаются и воспроизводятся нормально. Так же надо добавить плагин vlc-plugin-gnutls для подключения по протоколу. Соотношение сторон 3:4 по умолчанию, можно нажать "режим обрезки видео потока" но при закрытии приложения настройка не запоминается. Плохо что нужно тащить сам плеер без разрешения пользователя.

**Reply by trueromanus** [reply]
По поводу ошибок "тут мои полномочия все" эти ошибки на стороне libvlc происходят, природа их появления мне неизвестна.
"Так же надо добавить плагин vlc-plugin-gnutls для подключения по протоколу." - можете уточнить зачем нужен этот плагин?
"Соотношение сторон 3:4 по умолчанию, можно нажать "режим обрезки видео потока" но при закрытии приложения настройка не запоминается" - запишу в доделки. 
"Плохо что нужно тащить сам плеер без разрешения пользователя." - можно добавить описание, или вообще сделать два отдельных пакета с ним и без него, ну это как вариант :)

**Reply by Koi-foo** [reply]
Без плагина возникает ошибка подключения tls и плеер не может подключится к серверу.

**Reply by trueromanus** [reply]
Странно почему он не в составе зависимостей у самого VLC.
